README.md

```diff
@@ -2,101 +2,105 @@
- this piece of functionality is really boring.
+## Quick start
 
-## So? (should we continue struggling)
+Install
 
-This library was written to get rid of all this boring copy-paste-modify process.
-
-## How? (in which way does this library resolve the problem)
-
-Actions, reducer and tests are already written, just use it.
+```sh
+$ npm install redux-from-to
+```
 
-## Installation
+or
 
-1. Fetch the library from npm:
 ```sh
-you@your-computer /path/to/your/project $ yarn add redux-from-to
+$ yarn add redux-from-to
 ```
-2. Wrap your main reducer:
+
+Wrap your root reducer:
+
 ```js
+// store.js or wherever you create your store.
+
 import { createStore, applyMiddleware } from 'redux';
 import thunk from 'redux-thunk';
 import { wrapper } from 'redux-from-to';
 
-import mainReducer from './mainReducer'
+import rootReducer from './rootReducer'
 
 
 const store = createStore(
-  wrapper(mainReducer),
+  wrapper(rootReducer),
   applyMiddleware(thunk),
 )
 ```
-3. You're all set!
+
+You're all set!
+
 
 ## API
 
 ### fromTo(from, to, [through])
 
-Makes all the job for you. Dispatches REQUEST-FAILURE-SUCCESS, manages data in your store.
+_CAUTION: for now, `fromTo` works only with state stored as [Immutable](https://github.com/facebook/immutable-js/) iterables._
+
+Makes the API call, dispatches REQUEST and FAILURE or SUCCESS actions, and populates your store with response data.
 
 **Arguments:**
-1. `from` (Function): a function returning a Promise. Should resolve to dispatch SUCCESS or reject for FAILURE.
- Data, returned by this Promise, will be saved in the store.
-2. `to` (Object or Array<string>): target in your state, where you want to store information.
 
-`to` as object should contain 3 properties:
- - `request` (Array<string>): target to save boolean value if `from()` did not resolve/reject yet,
- - `failure` (Array<string>): target to save returned data on reject,
- - `success` (Array<string>): target to save returned data on resolve.
+`from` (Function)
+
+A function returning a Promise.
+
+`to` (Array<string> or Object)
+
+The path to the target location in your store for data returned by `from`.
+
+If `to` is an Array, `fromTo` will automatically create
 
-If `to` is an array of strings - it will be transformed into an object:
 ```js
 {
-  request: [...to, 'isRequesting'],
-  failure: [...to, 'error'],
-  success: [...to, 'data'],
+  isRequesting: (boolean),  // true if `from` has not resolved/rejected
+  error: (any),             // data returned by `from` rejecting, if any
+  data: (any),              // data returned by `from` resolving, if any
 }
 ```
 
-CAUTION: this library will work only with iterable properties of the state, created by [immutable](https://github.com/facebook/immutable-js/).
+at the target location.
+
+If you want to specify different locations for these three, pass an object like
+
+```js
+{
+  request: (Array<string>),
```

Comment:
Array can contain not only strings, but numbers and even booleans or objects afaik